sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -68,6 +66,32 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
             return Mono.just(ShouldRetryResult.noRetry());
         }
 
+        // when sessionTokenMismatchRetryAttempt == 0, this is a request routed
+        // to some possible local region (first region from preferredRegions) or
+        // first read-available (for read requests) region or first write-available
+        // (for write requests) region
+        // when sessionTokenMismatchRetryAttempt > 0, this is a request possibly routed to
+        // a different region which is not a local region
+        boolean isRequestForPossibleLocalRegion = request.requestContext.sessionTokenMismatchRetryAttempt == 0;
```

Comment:
imagine an account is configured with 3 regions, multi-master: A, B, C

I think the behavior we want is:
if regionSwitchHint == LOCAL_REGION_PREFERRED:
     A(5s), B(5s), C(5s)
if regionSwitchHint == REMOTE_REGION_PREFERRED:
    A(only loop through all replica once), B(only loop through replica once), C(only loop through once)


But with the current implementation:
if regionSwitchHint == LOCAL_REGION_PREFERRED:
     A(5s), B(5s), C(5s)
if regionSwitchHint == REMOTE_REGION_PREFERRED:
    A(only loop through all replica once), B(5s), C(5s)

Reply:
Discussed offline - we restrict retries to 1 per region when `REMOTE_REGION_PREFERRED.`